tests/tt_metal/tt_metal/unit_tests_fast_dispatch/command_queue/test_EnqueueTrace.cpp

```diff
@@ -192,29 +197,31 @@ TEST_F(CommandQueueFixture, EnqueueProgramDeviceCapture) {
 
         if (!has_trace) {
             // Program must be cached first
-            tid = BeginTraceCapture(this->device_, command_queue.id(), 2048);
+            tid = BeginTraceCapture(device, command_queue.id());
             EnqueueProgram(command_queue, simple_program, false);
-            EndTraceCapture(this->device_, command_queue.id(), tid);
+            EndTraceCapture(device, command_queue.id(), tid);
             has_trace = true;
         }
-        ReplayTrace(this->device_, command_queue.id(), tid, true);
+        ReplayTrace(device, command_queue.id(), tid, true);
 
         EnqueueReadBuffer(command_queue, output, trace_output_data.data(), true);
         if (has_eager) EXPECT_TRUE(eager_output_data == trace_output_data);
     }
 
     // Done
     Finish(command_queue);
-    ReleaseTrace(this->device_, tid);
+    ReleaseTrace(device, tid);
+    tt::tt_metal::CloseDevice(device);
 }
 
-TEST_F(CommandQueueFixture, EnqueueTwoProgramTrace) {
+TEST(TraceTests, EnqueueTwoProgramTrace) {
     // Get command queue from device for this test, since its running in async mode
-    CommandQueue& command_queue = this->device_->command_queue();
+    Device* device = tt::tt_metal::CreateDevice(0, 1, 0, 6144);
```

Comment:
apparently it doesn't need to match, but would be good to know where 6144 comes from

Reply:
The way we do the trace buffer calculation atm is by running the test, examining the assert that gets hit due to the trace region size being less than the trace size, and updating the size. 

We go through these messy steps because we can't really update the trace region size after the trace has run right now. This would require us to determine the DRAM footprint of the intermediate buffers created during the trace, so we can ensure that the intermediates created during trace don't clobber the trace buffer.

This seems like a logical next step for trace and we can work on this next.